src/app/DTT/business_locations/business_ask.js

```diff
@@ -0,0 +1,89 @@
+import { Box, Stack } from '@chakra-ui/react'
+import { isEmpty, lensProp, set } from 'ramda'
+import { useEffect, useState } from 'react'
+
+import RepeatableAsk from 'app/ASKS/repeatable-ask'
+import safelyParseJson from 'utils/helpers/safely-parse-json'
+import { useIsMobile } from 'utils/hooks'
+import AddressPicker from '../address/address_picker'
+import Text from '../text'
+
+const BusinessAsk = ({
+  questionCode,
+  onSendAnswer,
+  data,
+  regexPattern,
+  errorMessage,
+  parentCode,
+  placeholderName,
+  attributeCode,
+  targetCode,
+  clientId,
+}) => {
+  const isMobile = useIsMobile()
+  const [value, setValue] = useState(safelyParseJson(data?.value), {})
+
+  const textValue = JSON.stringify(value?.departments || [''])
+  const addressValue = value?.address || ''
+
+  const textData = set(lensProp('value'), textValue)(data)
+  const addressData = set(lensProp('value'), addressValue)(data)
+
+  const onAddressUpdate = answer => {
+    setValue(set(lensProp('address'), answer), value)
+  }
+
+  const onTextUpdate = answer => {
+    setValue(set(lensProp('departments'), answer), value)
+  }
+
+  useEffect(() => {
+    if (!isEmpty(value?.departments || '') && !isEmpty(value?.address || '')) {
+      onSendAnswer(value)
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value])
```

Comment:
We can, declare the department and value varibale first as 
let departments = value?.departments
let address = value?.address
and then check if these two changes in the dependency array, because value is an object here, whose reference will change everytime it is rendered.